tpm2/seal_test.go

```diff
@@ -310,248 +256,17 @@ func (s *sealSuite) TestProtectKeyWithTPMNoPCRPolicyCounterHandle(c *C) {
-				"model":        "fake-model",
-				"grade":        "secured",
-			}, "Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij")},
-		AuthKey: authKey})
-}
-
-type testProtectKeysWithTPMData struct {
-	n      int
-	params *ProtectKeyParams
-}
-
-func (s *sealSuite) testProtectKeysWithTPM(c *C, data *testProtectKeysWithTPMData) {
-	var keys []secboot.DiskUnlockKey
-	for i := 0; i < data.n; i++ {
-		key := make(secboot.DiskUnlockKey, 32)
-		rand.Read(key)
-
-		keys = append(keys, key)
-	}
-
-	protectedKeys, authKey, err := ProtectKeysWithTPM(s.TPM(), keys, data.params)
-	c.Check(err, IsNil)
-	c.Check(protectedKeys, HasLen, data.n)
-
-	policyAuthPublicKey, err := NewPolicyAuthPublicKey(authKey)
-	c.Assert(err, IsNil)
-
-	var pcrPolicyCounterPub *tpm2.NVPublic
-	var pcrPolicySequence uint64
-	if data.params.PCRPolicyCounterHandle != tpm2.HandleNull {
-		index, err := s.TPM().CreateResourceContextFromTPM(data.params.PCRPolicyCounterHandle)
-		c.Assert(err, IsNil)
-
-		pcrPolicyCounterPub, _, err = s.TPM().NVReadPublic(index)
-		c.Check(err, IsNil)
-
-		pcrPolicySequence, err = s.TPM().NVReadCounter(index, index, nil)
-		c.Check(err, IsNil)
-	}
-
-	expectedPolicyData, expectedPolicyDigest, err := NewKeyDataPolicy(tpm2.HashAlgorithmSHA256, policyAuthPublicKey, pcrPolicyCounterPub, pcrPolicySequence)
-	c.Assert(err, IsNil)
-
-	for i, k := range protectedKeys {
-		for _, model := range data.params.AuthorizedSnapModels {
-			ok, err := k.IsSnapModelAuthorized(authKey, model)
-			c.Check(err, IsNil)
-			c.Check(ok, testutil.IsTrue)
-		}
-
-		skd, err := NewSealedKeyData(k)
-		c.Assert(err, IsNil)
-		c.Check(skd.Validate(s.TPM().TPMContext, authKey, s.TPM().HmacSession()), IsNil)
-
-		c.Check(skd.Version(), Equals, uint32(3))
-		c.Check(skd.PCRPolicyCounterHandle(), Equals, data.params.PCRPolicyCounterHandle)
-
-		c.Check(skd.Data().Public().NameAlg, Equals, tpm2.HashAlgorithmSHA256)
-		c.Check(skd.Data().Public().AuthPolicy, DeepEquals, expectedPolicyDigest)
-		c.Check(skd.Data().Policy().(*KeyDataPolicy_v3).StaticData, tpm2_testutil.TPMValueDeepEquals, expectedPolicyData.(*KeyDataPolicy_v3).StaticData)
-
-		keyUnsealed, authKeyUnsealed, err := k.RecoverKeys()
-		c.Check(err, IsNil)
-		c.Check(keyUnsealed, DeepEquals, keys[i])
-		c.Check(authKeyUnsealed, DeepEquals, authKey)
-	}
-
-	if data.params.AuthKey != nil {
-		c.Check(authKey, DeepEquals, data.params.AuthKey)
-	}
-
-	if data.params.PCRProfile != nil {
-		// Verify that the keys are sealed with the supplied PCR profile by changing
-		// the PCR values.
-		_, err := s.TPM().PCREvent(s.TPM().PCRHandleContext(23), []byte("foo"), nil)
-		c.Check(err, IsNil)
-
-		for _, k := range protectedKeys {
-			_, _, err = k.RecoverKeys()
-			c.Check(err, ErrorMatches, "invalid key data: cannot complete authorization policy assertions: cannot execute PCR assertions: "+
-				"cannot execute PolicyOR assertions: current session digest not found in policy data")
-		}
-	}
-}
-
```

Comment:
explanation PR comments  or pointers to relevant commit messages if they exist of why the various tests here have been dropped would be good? some are obviously tied to model auth which is gone, but for some is less obvious why they are gone

Reply:
It looks like these are dropped because there's not an equivalent API that creates multiple keys in a single call, as we aren't sharing PCR policies anymore.